File: bubblesort/prog.rb
#!/bin/ruby

n = gets.strip.to_i
a = gets.strip.split(' ').map(&:to_i)

total_swaps = 0
n.times do |i|
  swaps = 0
  (n-1).times do |j|
    if a[j] > a[j+1]
       a[j], a[j+1] = a[j+1], a[j]
       swaps += 1
    end
  end
  total_swaps += swaps
  break if swaps == 0
end

puts "Array is sorted in #{total_swaps} swaps."
puts "First Element: #{a.first}"
puts "Last Element: #{a.last}"

File: bubblesort/prog.swift
import Foundation

let n = Int(readLine()!)!
var a = readLine()!.components(separatedBy: " ").map{ Int($0)! }

var total_swaps = 0
for i in 0..<n {
  var swaps = 0
  for j in 0..<n-1 {
    if a[j] > a[j+1] {
      swap(&a[j],&a[j+1])
      swaps+=1
    }
  }
  total_swaps += swaps
  if swaps == 0 { break }
}

print("Array is sorted in \(total_swaps) swaps.")
print("First Element: \(a.first!)")
print("Last Element: \(a.last!)")
